src/services/ftp/backend.rs

```diff
@@ -301,95 +269,96 @@ impl Accessor for Backend {
 
     async fn read(&self, args: &OpRead) -> Result<BytesReader> {
         let path = args.path();
-        let mut guard = self.client.lock().await;
+
+        let mut ftp_stream = self.ftp_connect(Operation::Read).await?;
 
         if let Some(offset) = args.offset() {
-            guard.resume_transfer(offset as usize).map_err(|e| {
-                other(ObjectError::new(
-                    Operation::Read,
-                    path,
-                    anyhow!("resume transfer request: {e:?}"),
-                ))
-            })?;
+            ftp_stream
+                .resume_transfer(offset as usize)
+                .await
+                .map_err(|e| {
```

Comment:
Do we need to handle `NotFound` here?

Reply:
Well we could just return the default metadata Instance here. I changed in my newest commit.